crates/sui-types/src/coin.rs

```diff
@@ -117,3 +124,72 @@ pub struct TreasuryCap {
     pub id: UID,
     pub total_supply: Supply,
 }
+
+pub fn merge_coins_to_first_coin(coins_to_merge: &mut Vec<Coin>) {
+    if coins_to_merge.is_empty() {
+        return;
+    };
+    let total_coins = coins_to_merge.iter().fold(0, |acc, c| acc + c.value());
```

Comment:
Much better now but still can be made clearer:
if you make merge and split a member function of `Coin`, the combination of balance should happen inside there, instead of accumulating the total and creating new balance explicitly here